File: cpp/n231_TwosMutiple.cpp
// https://leetcode-cn.com/problems/power-of-two/
#include <iostream>
#include <vector>
using namespace std;

class Solution{
public:
    bool isPowerofTwo(int n){
        if (n <= 0) return false;
        while (n % 2 == 0) n = n / 2;
        return (n == 1);
    }
};

int main(){
    Solution solu;
    cout << solu.isPowerofTwo(64) << endl;
}

File: python/n231_TwosMutiple.py
# https://leetcode-cn.com/problems/power-of-two/
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n <= 0: return False
        if n == 1: return True
        while n > 1:
            if n % 2 != 0:
                return False
            n = n // 2
        return True
        
    def isPowerOfTwo_a(self, n: int) -> bool:
        if n <= 0: return False
        return n == n & (-n)


if __name__ == "__main__":
    solu = Solution()
    print(solu.isPowerOfTwo(64))
    print(solu.isPowerOfTwo_a(64))